test/test_autograd.py

```diff
@@ -1624,6 +1624,19 @@ def test_sparse_gather_x_scalar(self):
     def test_sparse_gather_both_scalar(self):
         self._test_sparse_gather((), (), 0)
 
+    # autograd tests via common_method_invocations don't allow input tensors to
+    # be sparse (RuntimeError: gradcheck expects all tensor inputs are dense when
+    # check_sparse_nnz is set to False.)
+    def test_sparse_mask_autograd(self):
+        for device in ['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']:
+            tensor = torch.randn(3, requires_grad=True, device=device)
+            mask = torch.ones(3, device=device)
+            mask[1] = 0
+            mask = mask.to_sparse()
+            converted = tensor.sparse_mask(mask).to_dense()
+            converted.sum().backward()
+            self.assertEqual(tensor.grad, mask.to_dense())
```

Comment:
does gradcheck / gradgradcheck not work here?

Reply:
I think @nairbv said gradcheck / gradgradcheck can't handle sparse tensors (as autograd leafs).

But it could be good to run gradcheck on `tensor.sparse_mask(mask.to_sparse())` (where `tensor`, `mask` are dense tensors, if that works.

Reply:
ya, I think we've run into this before and it can be worked around.

Reply:
sparse_mask didn't allow for a dense mask